test/jdk/javax/swing/JFileChooser/bug4943900.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2004, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4943900
+ * @summary Tests that FileFilter combo box is shown in FileChooser
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException
+ * @run main/manual bug4943900
+ */
+
+import java.io.File;
+import javax.swing.JComponent;
+import javax.swing.JFileChooser;
+import javax.swing.UIManager;
+import javax.swing.filechooser.FileFilter;
+import jtreg.SkippedException;
+
+public class bug4943900 {
+    private static final String INSTRUCTIONS = """
+        1. When the test runs a JFileChooser will be displayed.
+
+        2. Ensure that there is a Filter combo box with these two items:
+         - Text Files (*.txt)  [must be selected when the dialog opens]
+         - All Files
+
+        3. Leave the "Text files" item selected and check that the
+        filter works: only *.txt files can appear in the files list
+        You can navigate directories in the FileChooser and find one
+        that contains some *.txt files to ensure they are shown in
+        the file list.
+
+        4. Try switching the filters and ensure that the file list
+        is updated properly.
+
+        If the FileFilter works correctly, press PASS else press FAIL.
```

Comment:
I wonder why so many people use capitals for *Pass* and *Fail*…

Reply:
Probably for emphasis

Reply:
All the documentation guides recommend using the same spelling as used on the UI controls themselves. If possible, the control names should be in bold.